src/Cryptol/Eval/Value.hs

```diff
@@ -105,6 +113,10 @@ data EvalOpts = EvalOpts
 data GenValue sym
   = VRecord !(RecordMap Ident (SEval sym (GenValue sym))) -- ^ @ { .. } @
   | VTuple ![SEval sym (GenValue sym)]              -- ^ @ ( .. ) @
+  | VEnum !(SInteger sym) !(IntMap (ConValue sym))
+    -- ^ @Just 2@, the integer is the tag (e.g., `Just`), and the map contains
+    -- the fields (e.g., `{ 0 -> ("Just",2) }`.
```

Comment:
The "integer is the tag" part is a bit confusing, since there are _two_ integers here: the tag (`0`) and the field of the constructor (`2`). Perhaps this would be clearer if you changed `2` to, say, `()`.